controllers/imageupdateautomation_controller.go

```diff
@@ -319,7 +319,7 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		if err == errNoChanges {
 			debuglog.Info("no changes made in working directory; no commit")
 			statusMessage = "no updates made"
-			if lastCommit, lastTime := auto.Status.LastPushCommit, auto.Status.LastPushTime; lastCommit != "" {
+			if lastCommit, lastTime := auto.Status.LastPushCommit, auto.Status.LastPushTime; lastCommit != "" && lastTime != nil {
 				statusMessage = fmt.Sprintf("%s; last commit %s at %s", statusMessage, lastCommit[:7], lastTime.Format(time.RFC3339))
```

Comment:
`lastCommit[:7]` will panic if lastCommit string is not as long as it assumes it is (e.g. at least 7 characters long). Therefore the if has to confirm that assumption.

Reply:
Good catch, I hadn't picked up on this.